pkg/certificate/daemon.go

```diff
@@ -12,41 +12,51 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package identity
+package certificate
 
 import (
+	"context"
 	"fmt"
-	"math/rand"
 	"os"
 	"path/filepath"
 	"strings"
+	"sync"
 	"time"
 
 	"github.com/AthenZ/k8s-athenz-sia/v3/pkg/config"
-	"github.com/AthenZ/k8s-athenz-sia/v3/pkg/token"
+	"github.com/AthenZ/k8s-athenz-sia/v3/pkg/daemon"
 	"github.com/AthenZ/k8s-athenz-sia/v3/third_party/log"
 	"github.com/AthenZ/k8s-athenz-sia/v3/third_party/util"
 	"github.com/cenkalti/backoff"
 )
 
-func Certificated(idConfig *config.IdentityConfig, stopChan <-chan struct{}) (error, <-chan struct{}) {
+type certService struct {
+	shutdownChan chan struct{}
+	shutdownWg   sync.WaitGroup
 
-	if stopChan == nil {
-		panic(fmt.Errorf("Certificated: stopChan cannot be empty"))
+	idCfg   *config.IdentityConfig
+	handler *identityHandler
+	run     func() error
+}
+
+func New(ctx context.Context, idCfg *config.IdentityConfig) (daemon.Daemon, error) {
+	if ctx.Err() != nil {
+		log.Info("Skipped certificate provider initiation")
+		return nil, nil
 	}
 
-	if idConfig.ProviderService == "" {
-		log.Infof("Certificate provisioning is disabled with empty options: provider service[%s]", idConfig.ProviderService)
+	if idCfg.ProviderService == "" {
```

Comment:
@WindzCUHK I also see that this change is unnecessary.

Reply:
- revert in commit: https://github.com/AthenZ/k8s-athenz-sia/pull/100/commits/adf7ade47db7b8736c3ad7bd49020c181680e05d
- split as PR: https://github.com/AthenZ/k8s-athenz-sia/pull/108